ddl/partition.go

```diff
@@ -112,6 +112,26 @@ func checkPartitionNameUnique(tbInfo *model.TableInfo, pi *model.PartitionInfo)
 	return nil
 }
 
+func checkPartitionFuncValid(expr ast.ExprNode) error {
+	switch v := expr.(type) {
+	case *ast.CaseExpr:
```

Comment:
The expression with ` + - * ` arithmetic can inside.

Reply:
done.PTAL